cmd/bucket-notification-handlers.go

```diff
@@ -288,6 +296,11 @@ func (l *listenChan) waitForListener(w http.ResponseWriter) {
 
 // ListenBucketNotificationHandler - list bucket notifications.
 func (api objectAPIHandlers) ListenBucketNotificationHandler(w http.ResponseWriter, r *http.Request) {
+	if api.gateway {
+		writeErrorResponse(w, ErrNotImplemented, r.URL)
+		return
+	}
+
```

Comment:
you can let the underlying object layer return the error if it does not implement this call.

Reply:
as discussed, for the time being this check is being left as is. Can be removed after notifications are moved to object layer and a SetNotificationConfig method is in place.